appengine/handleStatus.go

```diff
@@ -196,3 +208,20 @@ func isAuthorizedSubdomain(domain string, authorized map[string]struct{}) bool {
 		domain = domain[i+1 : len(domain)]
 	}
 }
+
+func selfTest(c context.Context, r *http.Request) error {
+	// Self test ACME challenge prefix is correctly mapped to us.
+	url := *r.URL
+	url.Path = selfTestPrefix
+	client := urlfetch.Client(c)
+	resp, err := client.Get(url.String())
+	if err != nil {
+		log.Errorf(c, "Self-test for ACME challenge path failed: %v", err)
```

Comment:
Consider moving the log.Errorf to the caller:

    log.Errorf(c, "Self-test for ACME challenge path failed: %v", err)

and then just `return err` here, and `return fmt.Errorf("Expected self-test response 418 but was: %d", resp.StatusCode)` below.